### DISCUSSION TITLE
require is not defined in mdx

### DISCUSSION THREAD

**Discussion by D-Thrane** [storybookjs/storybook#23540] [Help]
I'm Running latest storybook 7 on vite. I have a simpel MDX. But I get a error

require is not defined
ReferenceError: require is not defined
    at http://localhost:6006/node_modules/react/jsx-runtime.js?v=e0a9ac34:6:3
    
```
import { Meta, Story } from '@storybook/addon-docs/blocks';

<Meta title="Hello World" parameters={{ docs: { page: null } }} />

# Hello, World!

This is a simple example of an MDX file in Storybook.

<Story name="Default">
  {() => <div>Hello, World!</div>}
</Story>
```

**Comment by joshuaaron** [comment]
Hey @D-Thrane - having the exact same issue here and can't find a solution. Did you ever work this out?

**Reply by valentinpalkovic** [reply]
Do you have `"type": "module"` set in your package.json? The issue might be ESM related

**Reply by joshuaaron** [reply]
@valentinpalkovic  I did yes. Removing it doesn't seem to help unfortunately

`*.stories.mdx` throw this error, however `*.stories.tsx` seem to work fine. Any idea? Let me know what type of config/files you might need to see to help

**Reply by valentinpalkovic** [reply]
Can you provide a simple reproduction? After I am back from vacation I try to take a look.
